docs/software_requirements/interrupts.sdoc

```diff
@@ -6,57 +6,228 @@ REQ_PREFIX: ZEP-
 IMPORT_FROM_FILE: software_requirements.sgra
 
 [REQUIREMENT]
-UID: ZEP-58
+UID: ZEP-13001
 STATUS: Draft
 TYPE: Functional
 COMPONENT: Interrupts
-TITLE: Service routine for handling interrupts (ISR)
+TITLE: Installing static IRQ service routines (ISR).
 STATEMENT: >>>
-The Zephyr RTOS shall provide support a service routine for handling interrupts (ISR).
+Zephyr RTOS shall provide a mechanism to initialize a static IRQ service routine (ISR),
+providing all parameters needed to configure the hardware and software.
+The static IRQ shall be initially disabled.
 <<<
+RELATIONS:
+- TYPE: Parent
+  VALUE: ZEP-39
 
 [REQUIREMENT]
-UID: ZEP-59
+UID: ZEP-13002
 STATUS: Draft
 TYPE: Functional
 COMPONENT: Interrupts
-TITLE: Multi-level interrupts
+TITLE: Installing low-power and low-latency direct IRQ service routines (ISR).
 STATEMENT: >>>
-The Zephyr RTOS shall support multi-level preemptive interrupt priorities, when supported by hardware. Note: detailed analysis to demonstrate non interference will be needed here.
+Zephyr RTOS shall provide a mechanism to initialize an direct IRQ handler,
+providing all parameters needed to configure the hardware and software.
+
+The direct IRQ shall be initially disabled.
 <<<
+USER_STORY: >>>
+As the developer of low-power and low-latency applications, I need to implement ISRs 
+that avoid the normal interrupt and power management overhead.
+<<<
+RELATIONS:
+- TYPE: Parent
+  VALUE: ZEP-39
 
 [REQUIREMENT]
-UID: ZEP-60
+UID: ZEP-13003
 STATUS: Draft
 TYPE: Functional
 COMPONENT: Interrupts
-TITLE: Associating application code with interrupts
+TITLE: Direct ISR, Platform Specific helper macros.
 STATEMENT: >>>
-The Zephyr RTOS shall provide an interface for associating application code with specific interrupts. (CLARIFY: Can it be a deferred procedure call at interrupt context? Would be different requirement)
+Zephyr RTOS shall provide macros supporting development of direct ISRs.
+ * Minimal header instructions
+ * Minimal footer instructions
+ * Low Power termination
 <<<
+RELATIONS:
+- TYPE: Parent
+  VALUE: ZEP-39
 
 [REQUIREMENT]
-UID: ZEP-61
+UID: ZEP-13004
 STATUS: Draft
 TYPE: Functional
 COMPONENT: Interrupts
-TITLE: Enabling interrupts
+TITLE: Installing dynamic IRQ service routines (ISR).
 STATEMENT: >>>
-The Zephyr RTOS shall provide mechanisms to enable interrupts.
+Zephyr RTOS shall provide a mechanism to initialize a dynamic IRQ service routine, 
+providing all parameters needed to configure the hardware and software.
+The direct IRQ shall be initially disabled.
+<<<
+RELATIONS:
+- TYPE: Parent
+  VALUE: ZEP-39
+
+[REQUIREMENT]
+UID: ZEP-13005
+STATUS: Draft
+TYPE: Functional
+COMPONENT: Interrupts
+TITLE: Uninstalling dynamic IRQ service routines (ISR).
+STATEMENT: >>>
+Zephyr RTOS shall provide a mechanism to uninstall a dynamic ISR and reset the IRQ hardware. 
```

Comment:
Does this really rests the IRQ hardware?
In my opinion the hardware reset should be removed here because as far as I understand the functionality it only disconnects the dynamic ISR is removed from the association with the irq itself.

Reply:
Right!  I'll fix that.

Reply:
Done